tests/test_flow/test_reloading.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+import tempfile
+import sys
+import time
+
+
+@pytest.fixture(scope="function")
+def make_tmp_flow_module():
+    # We want to be able to do import statement on the temporary module, so we
+    # create a temporary directory here and add it to sys.path.
+    tmp_dir = tempfile.TemporaryDirectory()
+    sys.path.insert(0, tmp_dir.name)
+
+    def _make_tmp_flow_module(var_name, var_value):
+        with open(f"{tmp_dir.name}/tmp_flow_module.py", "w") as tmp_file:
+            tmp_file.write(
+                f"""
+import bionic as bn
+builder = bn.FlowBuilder('flow')
+builder.assign('x', {var_value})
+
+@builder
+def {var_name}(x):
+    '''documentation for {var_name}'''
+    return 2*x
+
+flow = builder.build()
+            """
+            )
+            tmp_file.flush()
+        # Needed to ensure that the file is loaded properly
+        # TODO: any way to avoid this?
+        time.sleep(1)
+
+    yield _make_tmp_flow_module
+
+    # Ensure that the temporary module is not loaded by a future test
+    sys.path.remove(tmp_dir.name)
+    # Remove module caching
+    sys.modules.pop("tmp_flow_module", None)
+
+
+def check_results(flow, results_map):
+    for k, v in results_map.items():
+        assert flow.get(k) == v
+        assert getattr(flow.get, k)() == v
+
+    cache_entities = [entry.entity for entry in flow.cache.get_entries()]
+    for k in results_map.keys():
+        assert k in cache_entities
+
+
+def test_reloading(make_tmp_flow_module):
+    make_tmp_flow_module("y", 1)
+
+    from tmp_flow_module import flow
+
+    check_results(flow, {"x": 1, "y": 2})
+
+    make_tmp_flow_module("z", 2)
+    new_flow = flow.reloading()
+    check_results(new_flow, {"x": 2, "z": 4})
```

Comment:
Since we don't mutate the existing flow, can you also check that the results of `flow` didn't change after reloading it?

Reply:
Updated the check methods so that it will remember and run previous checks if we are not reloading in place.